src/mgmt/srv_drpc.c

```diff
@@ -990,55 +1046,65 @@ add_props_to_resp(daos_prop_t *prop, Mgmt__PoolGetPropResp *resp)
 {
 	Mgmt__PoolProperty	**resp_props;
 	struct daos_prop_entry	*entry;
-	int			 i, rc = 0;
+	int			 i, rc = 0, valid_prop_nr = 0, j = 0;
 
 	if (prop == NULL || prop->dpp_nr == 0)
 		return 0;
 
-	D_ALLOC_ARRAY(resp_props, prop->dpp_nr);
+	for (i = 0; i < prop->dpp_nr; i++) {
+		entry = &prop->dpp_entries[i];
+		if (!(entry->dpe_flags & DAOS_PROP_ENTRY_NEGATIVE))
```

Comment:
Maybe we should have an inline helper function called something like `prop_is_set` for checking this flag. Looks like we do this in a number of places in C code now.